### PR TITLE
OPSEXP-1990: refactor zip builder script

### PR DESCRIPTION
**Pull request by alxgomz**
Ref: OPSEXP-1990

Tested build script in: https://github.com/Alfresco/alfresco-ansible-deployment/actions/runs/4556555199/jobs/8037061270

### CODE REVIEW

**Review by gionn — CHANGES_REQUESTED** [review]
[test run](https://github.com/Alfresco/alfresco-ansible-deployment/actions/runs/4556020022/jobs/8035833653#step:5:567) doesn't look good

can you also add a workflow_dispatch, have a fallback version and skip the maven upload? so we have a consistent way to test it without any major downside

**Comment by alxgomz** [comment]
> 

I linked the wrong build that is actually fixed (PR desc updated). Also added the workflow dispatch

**Review by slohe1 — DISMISSED** [review]

**Review by gionn — CHANGES_REQUESTED** [review]
.envrc can be removed

File: scripts/generate-zip.sh [hunk 1]
```diff
@@ -20,12 +20,11 @@ echo "Generating ZIP file..."
 find "$ARTIFACT_NAME" \( -type f \
 	! -path "${ARTIFACT_NAME}/.*" \
 	! -path "${ARTIFACT_NAME}/tests/*" \
-	! -path "${ARTIFACT_NAME}/molecule/*" \
+	! -path "${ARTIFACT_NAME}/*/molecule/*" \
```
Comment (gionn) [inline scripts/generate-zip.sh:23]: that will not match anymore the molecule in the root

**Review by gionn — CHANGES_REQUESTED** [review]

**Pull request merged**